### PR TITLE
feat(reports): read-only reports data layer (5 canned queries) (#211)

### PR DESCRIPTION
**Pull request by sydlexius**
## Summary

PR-A of #211 (Reports workspace), split into two PRs: this is the **read-only data layer**; the two-pane web UI follows in PR-B. Adds a new `internal/reports/` package with five canned, read-only report queries over the **existing** schema. **No schema migrations.**

The CodeRabbit Coding Plan on #211 was stale (it predated migration 018 + the #210 UI foundation) and assumed reports 3 & 4 needed new columns or placeholders. They do not: migration 018 already added `provider_lane` and `instrumental_result`, and the `provider_outcomes` table exists. All five reports are implemented for real over existing data.

## The five reports

Repository pattern (`Repo` wrapping `*sql.DB`, `New(db)`), one read-only `context`-aware method each:

1. **QueueSummary** - counts by status (pending/processing/done/failed/deferred), zero-count statuses reported as 0.
2. **RecentOutcomes(limit)** - last N `done` tracks, newest first (NULLs-last), with a SQL-derived result class from `output_paths` (`.lrc` -> synced, `.txt` -> unsynced-or-instrumental, miss-limit -> miss), guarded by `json_valid` against legacy/empty values.
3. **ProviderEffectiveness** - per-lane hits/misses/hit-rate from the worker-maintained `provider_outcomes` aggregate. **The hit-rate is attempt-weighted and APPROXIMATE** (a lane tried-but-lost to a later winning lane records neither a hit nor a miss), so it is labeled honestly and the UI is told to surface an "approximate" label. A true per-track hit-rate is tracked as follow-up **#282**.
4. **InstrumentalInventory** - tracks detected instrumental (`instrumental_result = 1`), distinguishing the detection-request flag (`detect_instrumental`) from the detection outcome, joined to `scan_results` for the file path.
5. **FailureAnalysis** - failed/deferred tracks grouped by status + normalized reason (`COALESCE(NULLIF(last_error,''),'unknown')`, consistent with the existing `CountFailuresByReason`).

## Constraints honored

- **Strictly read-only** - every method is a bare `SELECT` via `QueryContext`; zero write paths.
- **No migrations** - all five reports run on existing schema.
- Repository pattern + **real-SQLite** in-package integration tests (no mocks), mirroring `internal/cache/`.
- Patch coverage **80.13%** (threshold 70%).

## Test plan

- `make gate` green (build, race tests, patch coverage, golangci-lint, actionlint, govulncheck).
- `go test -race ./internal/reports/...` passes; whole-module `go build ./... && go vet ./...` clean.
- Tests assert concrete values + edge cases: NULLs-last ordering, miss-classification precedence, malformed/empty `output_paths` -> unknown, hit-rate divide-by-zero, instrumental tri-state with/without scan link, failure grouping by status + normalized reason, and error paths (closed DB, malformed timestamp).

## Notes

- Part of #211 (does **not** close it; PR-B - the two-pane UI - completes the feature).
- Follow-up: #282 (true per-track provider hit-rate).
- An independent hostile review found 0 MAJOR and no correctness/injection/read-only/scope defects; the 2 MINOR + 1 NIT it surfaced are fixed here.

## Summary by CodeRabbit

* **New Features**
  * New reporting suite now available: monitor queue status summary by work category, review recently completed outcomes with track metadata, measure provider effectiveness metrics by lane with hit rates, track instrumental music detections in inventory, and analyze failure patterns with categorized error reasons.

### CODE REVIEW

File: internal/reports/reports.go [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+// failed/deferred by migrations 001 + 012). Zero-count statuses are reported
+// as 0 rather than omitted.
+func (r *Repo) QueueSummary(ctx context.Context) (QueueSummary, error) {
+	var s QueueSummary
+	rows, err := r.db.QueryContext(ctx,
+		`SELECT status, COUNT(*) FROM work_queue GROUP BY status`)
+	if err != nil {
+		return QueueSummary{}, fmt.Errorf("reports: queue summary: %w", err)
+	}
+	defer rows.Close() //nolint:errcheck // read-only query; close error is not actionable
```
[the remark this answers is not in the document]
Comment (sydlexius) [reply internal/reports/reports.go]: Fixed in 4c19c6e: replaced all 5 `defer rows.Close() //nolint:errcheck` with `defer func() { _ = rows.Close() }()`, matching the prevailing repo idiom (internal/secrets/store.go:117, internal/library/repository.go:114, internal/queue/queue.go:1312). errcheck is now satisfied with no nolint directive.

File: internal/reports/reports.go [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+			o            RecentOutcome
+			completedAt  sql.NullString
+			providerLane sql.NullString
+			result       string
+		)
+		if err := rows.Scan(&o.Artist, &o.Title, &o.Album, &completedAt, &providerLane, &result); err != nil {
+			return nil, fmt.Errorf("reports: scan recent outcome: %w", err)
+		}
+		if completedAt.Valid && completedAt.String != "" {
+			t, perr := time.Parse(timeFormat, completedAt.String)
```
[the remark this answers is not in the document]
Comment (sydlexius) [reply internal/reports/reports.go]: Fixed in 4c19c6e: renamed `perr` -> `err` (all 3 references) per the AGENTS.md error-naming convention; it shadows the outer scope as that convention allows.

**Comment by sydlexius** [comment]
@coderabbitai status
